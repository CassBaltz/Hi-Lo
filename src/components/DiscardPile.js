import React from "react";
import { withStyles } from "@material-ui//styles";

const BlankCard = ({ classes, offSet, zIndex, }) => (
  <div
    className={classes.blankCard}
    style={{ bottom: offSet, right: offSet, zIndex, }}
  ></div>
);

const TopCard = ({ card, classes, offSet, zIndex }) => (
  <img
    className={classes.topCard}
    src={card.image}
    alt={card.code}
    style={{ bottom: offSet, right: offSet, zIndex, }}
  />
);

export const DiscardPile = ({ cardsShowing, classes }) => {
  const lastIndex = cardsShowing.length - 1;

  const discardPile = cardsShowing.map((card, index) => {
    const offSet = index * 3;
    if (index === lastIndex) {
      return (
        <TopCard
          key={card.code}
          card={card}
          classes={classes}
          offSet={offSet}
          zIndex={index}
        />
      );
    }

    return (
      <BlankCard
        key={card.code}
        offSet={offSet}
        classes={classes}
        zIndex={index}
      />
    );
  }
  );

  return (
    <div className={classes.root}>
      {discardPile}
      <p className={classes.count}>{discardPile.length}</p>
    </div>
  );
};

const styles = {
  root: {
    position: "relative",
    width: "70px",
    height: "98px",
    left: "100px",
  },
  topCard: {
    width: "70px",
    height: "98px",
    position: "absolute",
  },
  blankCard: {
    width: "70px",
    height: "98px",
    backgroundColor: "#FFF",
    borderRadius: "4px",
    border: "1px solid #ced4da",
    boxSizing: "border-box",
    position: "absolute",
  },
  count: {
    position: "absolute",
    fontSize: "18px",
    fontWeight: 700,
    bottom: "-45px",
    textAlign: "center",
    width: "70px",
    color: "#495057",
  },
};

export default withStyles(styles)(DiscardPile);
