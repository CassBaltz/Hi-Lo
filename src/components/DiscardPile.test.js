import React from "react";
import { DiscardPile } from "./DiscardPile";

const props = {
  cardsShowing: [
    {
      code: "AD",
      value: "ACE",
      image: "ace-url",
    },
  ],
  classes: {
    root: "root",
    topCard: "topCard",
    blankCard: "blankCard",
    count: "count",
  }
};

describe("<DiscardPile />", () => {
  it("should display correctly when there is only one card", () => {
    const singleCardPile = shallow(<DiscardPile {...props} />);
    expect(toJson(singleCardPile)).toMatchSnapshot();
  });
  it("should display correctly when there are multiple cards", () => {
    const newProps = {
      ...props,
      cardsShowing: [
        ...props.cardsShowing,
        {
          code: "KD",
          value: "KING",
          image: "King-url",
        }
      ]
    };
    const multipleCardPile = shallow(<DiscardPile {...newProps} />);
    expect(toJson(multipleCardPile)).toMatchSnapshot();
  });
});