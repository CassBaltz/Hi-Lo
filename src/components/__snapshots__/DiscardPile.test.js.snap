// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscardPile /> should display correctly when there are multiple cards 1`] = `
<div
  className="root"
>
  <BlankCard
    classes={
      Object {
        "blankCard": "blankCard",
        "count": "count",
        "root": "root",
        "topCard": "topCard",
      }
    }
    key="AD"
    offSet={0}
    zIndex={0}
  />
  <TopCard
    card={
      Object {
        "code": "KD",
        "image": "King-url",
        "value": "KING",
      }
    }
    classes={
      Object {
        "blankCard": "blankCard",
        "count": "count",
        "root": "root",
        "topCard": "topCard",
      }
    }
    key="KD"
    offSet={3}
    zIndex={1}
  />
  <p
    className="count"
  >
    2
  </p>
</div>
`;

exports[`<DiscardPile /> should display correctly when there is only one card 1`] = `
<div
  className="root"
>
  <TopCard
    card={
      Object {
        "code": "AD",
        "image": "ace-url",
        "value": "ACE",
      }
    }
    classes={
      Object {
        "blankCard": "blankCard",
        "count": "count",
        "root": "root",
        "topCard": "topCard",
      }
    }
    key="AD"
    offSet={0}
    zIndex={0}
  />
  <p
    className="count"
  >
    1
  </p>
</div>
`;
